### PR TITLE
Add documentation for RUSTC_EMIT

### PR DESCRIPTION
**Pull request by Zoxc**
This adds documentation for https://github.com/rust-lang/rust/pull/108365.

### CODE REVIEW

File: src/building/bootstrapping.md [hunk 1]
```diff
@@ -365,6 +365,16 @@ recompiling all dependencies.
...
 
+`RUSTC_EMIT` will cause `bootstrap` to pass `--emit` arguments on to `rustc`.
+You can specify which `--emit` formats which are used with a comma separated list prefixed with
+a crate name and `=`, for example: `rustc_arena=llvm-ir,asm`.
+You can also substitute `*` for the crate name to apply it to all crates.
+You can combine multiple such specifiers by seperating them with semi-colons like
+`rustc_arena=llvm-ir,asm;rustc_span=asm;*=llvm-bc`.
+The emitted files will be stored inside the `emit` directory in the target specific build directory.
+`RUSTC_EMIT` will not cause `bootstrap` to rebuild any crates so you may need to modify the crates
+you're interested in.
```
Comment (albertlarsan68) [inline src/building/bootstrapping.md:376]: Can you add an explanation on how to do this?

File: src/building/bootstrapping.md [hunk 2]
```diff
@@ -365,6 +365,16 @@ recompiling all dependencies.
 `CARGOFLAGS` will pass arguments to cargo itself (e.g. `--timings`). `CARGOFLAGS_BOOTSTRAP` and
 `CARGOFLAGS_NOT_BOOTSTRAP` work analogously to `RUSTFLAGS_BOOTSTRAP`.
 
+`RUSTC_EMIT` will cause `bootstrap` to pass `--emit` arguments on to `rustc`.
+You can specify which `--emit` formats which are used with a comma separated list prefixed with
+a crate name and `=`, for example: `rustc_arena=llvm-ir,asm`.
+You can also substitute `*` for the crate name to apply it to all crates.
+You can combine multiple such specifiers by seperating them with semi-colons like
+`rustc_arena=llvm-ir,asm;rustc_span=asm;*=llvm-bc`.
```
Comment (albertlarsan68) [inline src/building/bootstrapping.md:373]: Can you add the output formats please?

Comment (Zoxc) [reply src/building/bootstrapping.md:373]: Done.

**Review by albertlarsan68 — APPROVED** [review]

**Comment by albertlarsan68** [comment]
@jyn514 Can you merge this please?

File: src/building/bootstrapping.md [hunk 3]
```diff
@@ -365,6 +365,19 @@ recompiling all dependencies.
 `CARGOFLAGS` will pass arguments to cargo itself (e.g. `--timings`). `CARGOFLAGS_BOOTSTRAP` and
 `CARGOFLAGS_NOT_BOOTSTRAP` work analogously to `RUSTFLAGS_BOOTSTRAP`.
 
+`RUSTC_EMIT` will cause `bootstrap` to pass `--emit` arguments on to `rustc`.
+You can specify which `--emit` formats which are used with a comma separated list prefixed with
```
Comment (tshepang) [inline src/building/bootstrapping.md:369]:
```suggestion
You can specify which `--emit` formats are used with a comma separated list prefixed with
```

File: src/building/bootstrapping.md [hunk 4]
```diff
@@ -365,6 +365,19 @@ recompiling all dependencies.
...
+a crate name and `=`, for example: `rustc_arena=llvm-ir,asm`.
+You can also substitute `*` for the crate name to apply it to all crates.
+You can combine multiple such specifiers by seperating them with semi-colons like
+`rustc_arena=llvm-ir,asm;rustc_span=asm;*=llvm-bc`.
+Currently `rustc` supports `asm`, `llvm-bc`, `llvm-ir`,
```
Comment (tshepang) [inline src/building/bootstrapping.md:374]:
```suggestion
Currently, `rustc` supports `asm`, `llvm-bc`, `llvm-ir`,
```

File: src/building/bootstrapping.md [hunk 5]
```diff
@@ -365,6 +365,19 @@ recompiling all dependencies.
...
+`obj`, `metadata`, `link`, `dep-info`, and `mir` as `--emit` formats.
+The emitted files will be stored inside the `emit`
+directory in the target specific build directory.
```
Comment (tshepang) [inline src/building/bootstrapping.md:377]:
```suggestion
directory in the target-specific build directory.
```

File: src/building/bootstrapping.md [hunk 6]
```diff
@@ -365,6 +365,19 @@ recompiling all dependencies.
...
+`RUSTC_EMIT` will not cause `bootstrap` to rebuild any crates so you may need to modify
```
Comment (tshepang) [inline src/building/bootstrapping.md:378]:
```suggestion
`RUSTC_EMIT` will not cause `bootstrap` to rebuild any crates, so you may need to modify
```

**Comment by tshepang** [comment]
@albertlarsan68 this should only be merged after https://github.com/rust-lang/rust/pull/108365 is merged

**Comment by albertlarsan68** [comment]
Yeah, my bad.

**Pull request closed without merging**